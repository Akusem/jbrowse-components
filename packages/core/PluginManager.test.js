import { types, getSnapshot } from 'mobx-state-tree'
import PluginManager from './PluginManager'

describe('base data adapter', () => {
  it('properly propagates errors in feature fetching', async () => {
    const P = new PluginManager()
    P.addDisplayType(() => {
      return {
        name: 'KittyDisplay',
        configSchema: types.model({ hello: 'world' }),
      }
    })

    P.registerDisplayTypeForTrack({
      track: 'KittyTrack',
      display: 'KittyDisplay',
    })
    P.configure()

    expect(P.getElementType('display', 'KittyDisplay')).toMatchSnapshot()
    const res = P.pluggableDisplayTypesForTrack('KittyTrack')

    expect(getSnapshot(res.create({})).hello).toEqual('world')
  })
})
