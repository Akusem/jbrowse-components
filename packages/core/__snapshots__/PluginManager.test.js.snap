// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base data adapter properly propagates errors in feature fetching 1`] = `
Object {
  "configSchema": ModelType {
    "flags": 128,
    "identifierAttribute": undefined,
    "initializers": Array [],
    "isType": true,
    "name": "AnonymousModel",
    "named": [Function],
    "postProcessSnapshot": [Function],
    "preProcessSnapshot": [Function],
    "properties": Object {
      "hello": OptionalValue {
        "_defaultValue": "world",
        "_subtype": CoreType {
          "checker": [Function],
          "flags": 1,
          "initializer": [Function],
          "isType": true,
          "name": "string",
        },
        "isType": true,
        "name": "string",
        "optionalValues": Array [
          undefined,
        ],
      },
    },
    "propertyNames": Array [
      "hello",
    ],
    "props": [Function],
  },
  "name": "KittyDisplay",
}
`;
